packages/fluentui/react/src/components/Tree/Tree.tsx

```diff
@@ -191,135 +159,136 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
-          activeItemIds.push(sibling['id']);
-        }
-      });
+    const { id } = treeItemProps;
+    const { activeItemIds } = this.state;
+    const siblings = getSiblings(items, id);
 
-      if (hasSubtree(treeItemProps) && !this.isActiveItem(id)) {
-        activeItemIds.push(id);
+    siblings.forEach(sibling => {
+      if (hasSubtree(sibling) && !this.isActiveItem(sibling['id'])) {
+        activeItemIds.push(sibling['id']);
       }
+    });
 
-      this.setActiveItemIds(e, activeItemIds);
-
-      _.invoke(predefinedProps, 'onSiblingsExpand', e, treeItemProps);
+    if (hasSubtree(treeItemProps) && !this.isActiveItem(id)) {
+      activeItemIds.push(id);
     }
-  });
+
+    this.setActiveItemIds(e, activeItemIds);
+  };
 
   setActiveItemIds = (e: React.SyntheticEvent, activeItemIds: string[]) => {
-    _.invoke(this.props, 'onActiveItemIds', e, { ...this.props, activeItemIds });
+    _.invoke(this.props, 'onActiveItemIdsChange', e, { ...this.props, activeItemIds });
 
     this.setState({
       activeItemIds
     });
   };
 
+  contextValue: TreeItemRenderContextValue = {
+    onFocusParent: this.onFocusParent,
+    onSiblingsExpand: this.onSiblingsExpand,
+    onFocusFirstChild: this.onFocusFirstChild,
+    onTitleClick: this.onTitleClick
+  };
+
   renderContent(accessibility: ReactAccessibilityBehavior): React.ReactElement[] {
-    const { itemsForRender } = this.state;
     const { items, renderItemTitle } = this.props;
 
     if (!items) return null;
 
-    const renderItems = (items: ShorthandCollection<TreeItemProps>): React.ReactElement[] => {
-      return items.reduce((renderedItems: React.ReactElement[], item: ShorthandValue<TreeItemProps>) => {
-        const itemForRender = itemsForRender[item['id']];
-        const { elementRef, ...restItemForRender } = itemForRender;
+    const renderItems = (items: ShorthandCollection<TreeItemProps>, level = 1, parent?: string): React.ReactElement[] => {
+      return items.reduce((renderedItems: React.ReactElement[], item: ShorthandValue<TreeItemProps>, index: number) => {
+        const itemId = item['id'];
         const isSubtree = hasSubtree(item);
-        const isSubtreeExpanded = isSubtree && this.isActiveItem(item['id']);
+        const isSubtreeExpanded = isSubtree && this.isActiveItem(itemId);
+
+        if (!this.itemsRef.has(itemId)) {
+          this.itemsRef.set(itemId, React.createRef<HTMLElement>());
+        }
+
         const renderedItem = TreeItem.create(item, {
           defaultProps: () => ({
             accessibility: accessibility.childBehaviors ? accessibility.childBehaviors.item : undefined,
             className: Tree.slotClassNames.item,
             expanded: isSubtreeExpanded,
             renderItemTitle,
             key: item['id'],
-            contentRef: elementRef,
-            ...restItemForRender
-          }),
-          overrideProps: this.handleTreeItemOverrides
+            parent,
+            level,
```

Comment:
I want to refactor level, index, treeSize in a subsequent PR. I think it makes sense to just call them ariaPosInSet, ariaSetSize, ariaLevel since that's their only use. Opinions?

Reply:
`aria-level` in this case. But makes sense only if they are used only for attributes...